homeassistant/components/google_assistant/helpers.py

```diff
@@ -120,31 +120,29 @@ def async_disable_report_state(self):
             self._unsub_report_state()
             self._unsub_report_state = None
 
-    async def async_sync_entities(self):
+    async def async_sync_entities(self, agent_user_id: str):
         """Sync all entities to Google."""
         # Remove any pending sync
-        if self._google_sync_unsub:
-            self._google_sync_unsub()
-            self._google_sync_unsub = None
+        self._google_sync_unsub.pop(agent_user_id, lambda: None)()
 
-        return await self._async_request_sync_devices()
-
-    async def _schedule_callback(self, _now):
-        """Handle a scheduled sync callback."""
-        self._google_sync_unsub = None
-        await self.async_sync_entities()
+        return await self._async_request_sync_devices(agent_user_id)
 
     @callback
-    def async_schedule_google_sync(self):
+    def async_schedule_google_sync(self, agent_user_id: str):
         """Schedule a sync."""
-        if self._google_sync_unsub:
-            self._google_sync_unsub()
 
-        self._google_sync_unsub = async_call_later(
-            self.hass, SYNC_DELAY, self._schedule_callback
+        async def _schedule_callback(_now):
+            """Handle a scheduled sync callback."""
+            self._google_sync_unsub.pop(agent_user_id)
```

Comment:
If a sync is called, this might raise key error, so guard for it.

```suggestion
            self._google_sync_unsub.pop(agent_user_id, None)
```